Cython/Compiler/ExprNodes.py

```diff
@@ -14101,6 +14156,33 @@ def generate_result_code(self, code):
         ))
 
 
+class CTupleCastNode(CoercionNode):
+    """
+    Coerce between different CTuple types.
+    """
+
+    def __init__(self, arg, dst_type, env):
+        assert dst_type.is_ctuple
+        assert arg.type.is_ctuple
+        CoercionNode.__init__(self, arg)
+        self.type = dst_type
+        self.is_temp = 1
+        self.arg = arg
+
+    def generate_result_code(self, code):
+        self.recursive_assign(self.arg.result(), self.temp_code, self.arg.type, self.type, code)
+
+    def recursive_assign(self, src, dest, src_type, dest_type, code):
+        if dest_type.is_ctuple and dest_type != src_type:
+            for index, (member_src_type, member_dest_type) in enumerate(zip(src_type.components, dest_type.components)):
+                member_src = src + '.f' + str(index)
+                member_dest = dest + '.f' + str(index)
```

Comment:
This might be something that the type should be responsible for rather than an arbitrary external syntax node.

Reply:
Done